src/contracts/core/StrategyManager.sol

```diff
@@ -186,129 +183,38 @@ contract StrategyManager is
-            strategies[i] = stakerStrategyList[staker][index];
-            shares[i] = stakerStrategyShares[staker][strategies[i]];
-            strategyIndexes[i] = index;
-            unchecked {
-                ++i;
-            }
-        }
-        bytes32 queuedWithdrawal = _queueWithdrawal(staker, strategyIndexes, strategies, shares, staker);
-        return (strategies, shares, queuedWithdrawal);
+    function removeShares(
+        address staker,
+        IStrategy strategy,
+        uint256 shares
+    ) external onlyDelegationManager {
+        _removeShares(staker, 0, strategy, shares);
     }
 
-    /**
-     * @notice Called by a staker to queue a withdrawal of the given amount of `shares` from each of the respective given `strategies`.
-     * @dev Stakers will complete their withdrawal by calling the 'completeQueuedWithdrawal' function.
-     * User shares are decreased in this function, but the total number of shares in each strategy remains the same.
-     * The total number of shares is decremented in the 'completeQueuedWithdrawal' function instead, which is where
-     * the funds are actually sent to the user through use of the strategies' 'withdrawal' function. This ensures
-     * that the value per share reported by each strategy will remain consistent, and that the shares will continue
-     * to accrue gains during the enforced withdrawal waiting period.
-     * @param strategyIndexes is a list of the indices in `stakerStrategyList[msg.sender]` that correspond to the strategies
-     * for which `msg.sender` is withdrawing 100% of their shares
-     * @param strategies The Strategies to withdraw from
-     * @param shares The amount of shares to withdraw from each of the respective Strategies in the `strategies` array
-     * @param withdrawer The address that can complete the withdrawal and will receive any withdrawn funds or shares upon completing the withdrawal
-     * @return The 'withdrawalRoot' of the newly created Queued Withdrawal
-     * @dev Strategies are removed from `stakerStrategyList` by swapping the last entry with the entry to be removed, then
-     * popping off the last entry in `stakerStrategyList`. The simplest way to calculate the correct `strategyIndexes` to input
-     * is to order the strategies *for which `msg.sender` is withdrawing 100% of their shares* from highest index in
-     * `stakerStrategyList` to lowest index
-     */
-    function queueWithdrawal(
-        uint256[] calldata strategyIndexes,
-        IStrategy[] calldata strategies,
-        uint256[] calldata shares,
-        address withdrawer
-    ) external onlyWhenNotPaused(PAUSED_WITHDRAWALS) onlyNotFrozen(msg.sender) nonReentrant returns (bytes32) {
-        bytes32 queuedWithdrawal = _queueWithdrawal(msg.sender, strategyIndexes, strategies, shares, withdrawer);
-        delegation.decreaseDelegatedShares(msg.sender, strategies, shares);
-        return queuedWithdrawal;
+    function addShares(
+        address grantee,
```

Comment:
staker?

Reply:
I like the unity/consistency of language, will make the change

Reply:
I updated this to 'staker'. it's consistent with our other usage.